CMakeLists.txt

```diff
@@ -324,12 +325,11 @@ set(BUILD_LINK_WITH_GOLD ON CACHE BOOL "Use Gold linker for build if available")
 set(BUILD_LINK_WITH_LLD ON CACHE BOOL "Use LLD linker for build if available (overrides BUILD_LINK_WITH_GOLD)")
 
 # Build with LTO
-if (ENABLE_LTO AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+if (CMAKE_INTERPROCEDURAL_OPTIMIZATION AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
   if (BUILD_LINK_WITH_LLD)
     message(WARNING "LLD does not work with GCC's LTO object format, switching to Gold.")
     set(BUILD_LINK_WITH_LLD OFF)
   endif ()
-  add_cxx_compiler_option("-flto")
 endif ()
```

Comment:
So cmake adds the correct flags automatically, right? In that case, it mighyt handle the linker/compiler mismatch too, it would be worth testing.

Reply:
@vlstill p4c isn't using the CMake-standard way of specifying a linker either but is instead appending `-fuse-ld=lld` to the compiler command line, so it's not possible that CMake would autocorrect that.

(Not that it would anyway: I don't quite remember, but I think what it does is fail to configure if you specify interprocedural optimization with an incompatible linker type.  Not sure though.)